spec/namespaces.md

```diff
@@ -6,11 +6,16 @@ Using directives ([Using directives](namespaces.md#using-directives)) are provid
 
 ## Compilation units
 
-A *compilation_unit* defines the overall structure of a source file. A compilation unit consists of zero or more *using_directive*s followed by zero or more *global_attributes* followed by zero or more *namespace_member_declaration*s.
+A *compilation_unit* defines the overall structure of a source file. A compilation unit consists of zero or more *using_directive*s followed by zero or more *global_attributes* followed by a *compilation_unit_body*. A *compilation_unit_body* can either be a *single_line_namespace* or zero or more *namespace_member_declaration*s.
 
 ```antlr
 compilation_unit
-    : extern_alias_directive* using_directive* global_attributes? namespace_member_declaration*
+    : extern_alias_directive* using_directive* global_attributes? compilation_unit_body
```

Comment:
You'll want to integrate with top-level statements syntax: https://github.com/dotnet/csharplang/blob/master/proposals/csharp-9.0/top-level-statements.md#syntax

Reply:
Never mind, I see that you have a follow-up in OP. I think I like the latter proposal better. Allows to write:
```
// top-level code...

namespace NS;
class C
{
   ...
}
```